geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+If no listener is targeted (`sectionName`/`port` are unset) then the Route references all the listeners on the child Gateway. It MUST NOT attach
+to a listener on a parent or sibling Gateway.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: child-gateway
+    sectionName: metrics
+```
+
+Routes can only bind to listeners *directly* defined in the `Gateway` referenced.
+For instance, the following route referencing a listener defined in the parent `Gateway`, but attaching to the child `Gateway` is not valid.
+This will be reported in [status](#status-fields).
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: child-gateway
+    sectionName: section-in-parent
+```
+
+#### Merging Spec Fields
+
+##### Validation
+
+A parent and child MUST have the same `gatewayClassName`.
+This will be detected by the implementation and reported in [status](#status-fields).
+
+A child resource MUST not set any `spec.infrastructure` fields beyond `attachTo`, and cannot set `spec.address`.
+This can be validated in the CRD schema potentially using CEL.
+
+A parent resource MUST not set `spec.infrastructure.attachTo`.
+That is, we do not allow multiple tiers of Gateways chaining to each other; there is only a single parent with children.
+This can be validated in the CRD schema.
+
+A child resource cannot `attachTo` any Gateway resource that doesn't allow attachment (eg. it does not specify `spec.infrastructure.allowedChildren` for `Gateway`s).
+This will be detected by the implementation and reported in [status](#status-fields).
+
+##### Listeners
+
+Implementations MUST treat the "parent" Gateway as having the concatenated list of all listeners from itself and "child" Gateways.
+
+Validation of this list of listeners MUST behave the same as if the list were part of a single "parent" Gateway.
+
+eg.
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  infrastructure:
+    allowedChildren:
+      namespaces:
+        from: Same
+  gatewayClassName: example
+  listeners:
+  - name: HTTP
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: child-gateway
+spec:
+  gatewayClassName: example
+  infrastructure:
+    attachTo:
+      name: parent-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  listeners:
+  - name: metrics
+    port: 8080
+    protocol: HTTP
+```
+
+With this configuration, the realized "parent" Gateway should listen on port `80` and `8080`.
+
+###### Listener Precedence
+
+Gateway Listeners should be merged using the following precedence:
+- "parent" Gateway
+- "child" Gateway ordered by creation time (oldest first)
+- "child" Gateway ordered alphabetical by “{namespace}/{name}”.
```

Comment:
Does this mean that the oldest listeners from a set of child Gateways will "win" in a conflct of conflict listeners from the set of child Gateways?  And if all are created at the same name, then the ordering goes by alphabetical namespace/name?  So everything in one early-letter namespace wins over any other namespace?

Reply:
Yes. Let me number the list to make it obvious there's an order.

This is similar to how route matching works

https://github.com/kubernetes-sigs/gateway-api/blob/e971a8db22e70b77775622c5ed613c1bb5067e62/apis/v1/gateway_types.go#L304-L320

Reply:
updated GEP